packages/build/src/index.ts

```diff
@@ -38,29 +40,46 @@ if (require.main === module) {
       );
     }
 
-    if (command === 'trigger-release') {
-      await triggerRelease(process.argv.slice(3));
-    } else {
-      const config: Config = require(path.join(
-        __dirname,
-        '..',
-        '..',
-        '..',
-        'config',
-        'build.conf.js'
-      ));
+    switch (command) {
+      case 'trigger-release':
+        await triggerRelease(process.argv.slice(3));
+        break;
+      case 'update-cta':
+        const ctaConfig: UpdateCTAConfig = require(path.join(
+          __dirname,
+          '..',
+          '..',
+          '..',
+          'config',
+          'cta.conf.js'
+        ));
 
-      const cliBuildVariant = process.argv
-        .map((arg) => /^--build-variant=(.+)$/.exec(arg))
-        .filter(Boolean)[0];
-      if (cliBuildVariant) {
-        config.packageVariant = cliBuildVariant[1] as PackageVariant;
-        validatePackageVariant(config.packageVariant);
-      }
+        ctaConfig.isDryRun ||= process.argv.includes('--dry-run');
 
-      config.isDryRun ||= process.argv.includes('--dry-run');
+        await updateJsonFeedCTA(ctaConfig);
+        break;
```

Comment:
I think this is the only step in this approach that I don't fully follow – is the idea that we'd locally run the `update-cta` command? How would we get the credentials to access the relevant S3 bucket?

We're definitely moving towards a world in which less and less access will be given to individual developers' machines, so I don't think this is something we could easily do, and we may just need to fall back to a variant in which we update the CTA entries along with the version feed during releases only – I think that would be okay though, we're generally free to do mongosh releases whenever necessary.

(In the future, we'll need to make sure that even this kind of flow where we update an existing file in S3 as part of the releases is supported – see e.g. https://jira.mongodb.org/browse/WRITING-16730)

Reply:
I was hoping we could decouple releases from CTA updates. I had the following in mind - let me know how it sounds:
1. Updates are published when `cta.conf.js` is updated. This will have to go through the normal PR flow of accepting changes.
2. When the change is merged into `main`, a github action triggers that has access to the S3 credentials through environment secrets and runs the `update-cta` command.
3. The secrets are stored in an environment - let's say "production" - that will have protection rules configured, which means the deployment will be paused until the protection rules are satisfied (e.g. approval required by specified people/wait specified number of hours).

Reply:
> a github action triggers that has access to the S3 credentials through environment secrets and runs the update-cta command.

We can do that but I think we'd be going for Evergreen rather than GHA here? That's the only platform I'd really expect to be available to access release infrastructure in the long run. I think we'd encounter some resistance if we wanted to put downloads.mongodb.com secrets into another CI environment

Reply:
We could sort it out - my personal preference is towards GHA and we already have a handful of products being signed and released with secrets stored in GHA, as well as all our code being hosted by github. That being said, if there's a strong desire to only do that on evergreen, we could work with those limitations - the important part here is that the flow of publishing CTA updates happens on CI and in a somewhat automated and reviewable fashion.